packages/shared/src/components/modals/NewSquadModal.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { ReactElement, useState } from 'react';
+import { Modal } from './common/Modal';
+import ArrowIcon from '../icons/Arrow';
+import { Button } from '../buttons/Button';
+import { Squad } from '../../graphql/squads';
+import { SquadDetails } from '../squads/Details';
+import { SquadSelectArticle } from '../squads/SelectArticle';
+import { SquadReady } from '../squads/Ready';
+import {
+  createSquad,
+  ModalState,
+  SquadForm,
+  SquadStateProps,
+} from '../squads/utils';
+import { SquadComment } from '../squads/Comment';
+
+export const modalStateOrder = [
+  ModalState.Details,
+  ModalState.SelectArticle,
+  ModalState.WriteComment,
+  ModalState.Ready,
+];
+
+export type NewSquadModalProps = {
+  onRequestClose: () => void;
+  onPreviousState: () => void;
+};
+
+function NewSquadModal({
+  onPreviousState,
+  onRequestClose,
+}: NewSquadModalProps): ReactElement {
+  const [modalState, setModalState] = useState(ModalState.Details);
+  const [squad, setSquad] = useState<Squad>();
+  const changeState = (prev: boolean) => {
+    const index = modalStateOrder.findIndex((state) => state === modalState);
+    const changeTo = modalStateOrder[index + (prev ? -1 : 1)];
+    if (changeTo) setModalState(changeTo);
+    else if (prev) onPreviousState();
+    else onRequestClose();
+  };
+  const stepperWidth = () => {
+    const index =
+      modalStateOrder.findIndex((state) => state === modalState) + 1;
+    return (index / modalStateOrder.length) * 100;
+  };
+  const [form, setForm] = useState<Partial<SquadForm>>({});
+  const onNext = async (squadForm?: SquadForm) => {
+    const currentState = modalState;
+    changeState(false);
+    if (squadForm) setForm(squadForm);
+    if (ModalState.WriteComment !== currentState) return;
+    const newSquad = await createSquad(squadForm);
+    setSquad(newSquad);
+  };
+  const stateProps: SquadStateProps = {
+    modalState,
+    form,
+    setForm,
+    onNext,
+  };
+  return (
+    <Modal
```

Comment:
Should this not be the SteppedModal we already have?

Reply:
SteppedModal does not use the modal guidelines and still has not been updated. And it is only being used in Onboarding. I didn't want to use the old method

Reply:
Ah alright, thanks for explaining.
Is there anyway we can create a "new" stepped modal version?

I assume we'll be getting more and more "stepped" modals.

Reply:
Sure thing, let me work on that